src/tests/example.spec.js

```diff
@@ -1,27 +1,193 @@
+
+    // select random product index
+    const randomProductNumbers =  Math.floor(Math.random()*addToCartBtns.length) +1 
+
+    // creation an array of random number of products
+
+    let arrayOfProductRandomIndex = [];
+
+    for( let i = 0; i < randomProductNumbers; i++){
+        const randomFunction = async function(){
+            let randomProductIndex = Math.floor(Math.random()*addToCartBtns.length)+1;
+            // const arrayItemExisting = randomArrayOfProductItems.includes(randomProductIndex)
+            if(arrayOfProductRandomIndex.includes(randomProductIndex)){
+                randomFunction()
+            }else {
+                arrayOfProductRandomIndex.push(randomProductIndex)
+            }
+        }
+        randomFunction()
+    }
+
+    let sortedArrayOfProductRandomIndex =  arrayOfProductRandomIndex.sort()
+   
+    //Get information about products
+    const productsInformation = await inventoryPage.getProductsInformation();
+
+   //click on Add to card button
+
+    for(let i = 0; i < sortedArrayOfProductRandomIndex.length; i++) {
+        await inventoryPage.addItemToCartByInventoryItem(sortedArrayOfProductRandomIndex[i])
+    }
+
+    //click on shopping icon
+    await inventoryPage.shopingCart.click();
+
+    //Get information about products on ShoppingCardPage
+    const shoppingProductCardItems = await shopingCartPage.getCartProductsInformation()
+
+    for(let i = 0 ; i < sortedArrayOfProductRandomIndex.length; i++){
+   
+   
+    await expect(productsInformation[sortedArrayOfProductRandomIndex[i]-1].name).toEqual(shoppingProductCardItems[i].name);
+   
+    await expect(productsInformation[sortedArrayOfProductRandomIndex[i]-1].description).toEqual(shoppingProductCardItems[i].description);
+    await expect(productsInformation[sortedArrayOfProductRandomIndex[i]-1].price).toEqual(shoppingProductCardItems[i].price);
+        
+    }
+
+  });
+
+  test("Verify checkout information", async ({ inventoryPage, shopingCartPage, checkoutInfoPage , checkoutOverviewPage}) => {
+     //select all "Add to cards btns"
+     const addToCartBtns = await inventoryPage.inventoryItems.allTextContents();
+
+     // select random product index
+     const randomProductNumbers =  Math.floor(Math.random()*addToCartBtns.length) +1 
+ 
+     // creation an array of random number of products
+ 
+     let arrayOfProductRandomIndex = [];
+ 
+     for( let i = 0; i < randomProductNumbers; i++){
+         const randomFunction = async function(){
+             let randomProductIndex = Math.floor(Math.random()*addToCartBtns.length)+1;
+             // const arrayItemExisting = randomArrayOfProductItems.includes(randomProductIndex)
+             if(arrayOfProductRandomIndex.includes(randomProductIndex)){
+                 randomFunction()
+             }else {
+                 arrayOfProductRandomIndex.push(randomProductIndex)
+             }
+         }
+         randomFunction()
+     }
```

Comment:
Oh please do something with this code as well :) For me it looks like I need a cup of coffee, debugger and an hour to spot what is going on :)